Xamarin.Forms.Core/ResourceDictionary.cs

```diff
@@ -128,8 +185,13 @@ IEnumerator IEnumerable.GetEnumerator()
 			return _innerDictionary.GetEnumerator();
 		}
 
-		internal IEnumerable<KeyValuePair<string, object>> MergedResources {
-			get {
+		internal IEnumerable<KeyValuePair<string, object>> MergedResources
+		{
+			get
+			{
+				if (MergedDictionaries != null)
+					foreach (var r in MergedDictionaries.Reverse().SelectMany(x => x.MergedResources))
```

Comment:
Using a for loop on an ICollection, would result in using ElementAt, which is less efficient than just using a foreach loop.